woocommerce-gutenberg-products-block.php

```diff
@@ -16,22 +16,23 @@
 defined( 'ABSPATH' ) || die();
 
 define( 'WGPB_VERSION', '2.0.0-dev' );
-
-define( 'WGPB_DEVELOPMENT_MODE', true );
+define( 'WGPB_PLUGIN_FILE', __FILE__ );
+define( 'WGPB_ABSPATH', dirname( WGPB_PLUGIN_FILE ) . '/' );
 
 /**
  * Load up the assets if Gutenberg is active.
  */
 function wgpb_initialize() {
-	$files_exist = file_exists( plugin_dir_path( __FILE__ ) . '/build/featured-product.js' );
+	require_once plugin_dir_path( __FILE__ ) . 'assets/php/class-wgpb-block-library.php';
 
-	if ( $files_exist && function_exists( 'register_block_type' ) ) {
-		add_action( 'init', 'wgpb_register_blocks' );
-		add_action( 'init', 'wgpb_register_scripts' );
-		add_filter( 'block_categories', 'wgpb_add_block_category' );
-	}
+	// Remove core hooks in favor of our local feature plugin handlers.
+	remove_action( 'init', array( 'WC_Block_Library', 'register_blocks' ) );
+	remove_action( 'init', array( 'WC_Block_Library', 'register_assets' ) );
+	remove_filter( 'block_categories', array( 'WC_Block_Library', 'add_block_category' ) );
+	remove_action( 'admin_print_footer_scripts', array( 'WC_Block_Library', 'print_script_settings' ), 1 );
```

Comment:
Since this is fired on `woocommerce_loaded`, we know the core class has already been registered. We can remove the hooks here, and then in `WGPB_Block_Library` set up our local feature-plugin hooks. The `WGPB_Block_Library` class is copied over directly, so any new files we add in will be automatically brought along with the grunt sync process.